src/configurationProvider.ts

```diff
@@ -189,6 +189,44 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
     private isEmptyArray(configItems: any): boolean {
         return !Array.isArray(configItems) || !configItems.length;
     }
+
+    private async chooseMainClass(folder: vscode.WorkspaceFolder | undefined): Promise<IMainClassOption> {
+        const res = await resolveMainClass(folder ? folder.uri : undefined);
+        if (res.length === 0) {
+            vscode.window.showErrorMessage(
+                "Cannot find a class with the main method.");
+            return;
+        }
+        const pickItems = res.map((item) => {
+            let name = item.mainClass;
+            let details = `main class: ${item.mainClass}`;
+            if (item.projectName !== undefined) {
+                name += `<${item.projectName}>`;
+                details += ` | project name: ${item.projectName}`;
+            }
+            return {
+                description: details,
+                label: name,
+                item,
+            };
+        }).sort ((a, b): number => {
+            return a.label > b.label ? 1 : -1;
+        });
+        if (pickItems.length === 0) {
+            // user gives up the selection, quit sliently
+            return;
+        }
+        const selection = pickItems.length > 1 ?
+            await vscode.window.showQuickPick(pickItems, { placeHolder: "Select main class<project name>" })
+            : pickItems[0];
+        if (selection && selection.item) {
+            return selection.item;
+        } else {
+            vscode.window.showErrorMessage("Please specify the mainClass (e.g. [mymodule/]com.xyz.MainClass) in the launch.json.");
+            this.log("usageError", "Please specify the mainClass (e.g. [mymodule/]com.xyz.MainClass) in the launch.json.");
+            return undefined;
```

Comment:
Given that you have used `return ` in the previous side branch, use `return` directly here to keep code consistent.